dnadoh/synthesize.py

```diff
@@ -196,10 +218,41 @@ def make_weight(self, person, reference, individual):
         mean = self.WEIGHT_MEANS[person.gsex]
         std_dev = mean * self.WEIGHT_RSD
         person.weight = _truncate(random.gauss(mean, std_dev), 1)
+    
+    def make_household_id(self, person, reference, individual):
+        person.household_id = self.hh_id
 
     def _make_pid(self, i):
         """Create personal identifier."""
         return str(i).zfill(PID_WIDTH)
+    
+    def _random_household_size(self):
+        return np.random.choice(a = [key for key in self.HOUSEHOLD_SIZE_DISTRIB.keys()], 
+                                size = 1, 
+                                p = [val for val in self.HOUSEHOLD_SIZE_DISTRIB.values()])
+
+    def _init_household(self):
+        """Initializes household-related variables"""
+
+        # Select a household size weighted by specified probabilities
+        self.hh_size =  self._random_household_size()
+
+        # Create variables to track 'current' household ID and how many individuals we've generated for that household
+        self.hh_id = 0
+        self.hh_individuals = 0
+    
+    def _iter_household(self):
+        """Iterate household variables, track number of individuals per HH and move to next HH if required"""
+        self.hh_individuals += 1
+
+        if self.hh_individuals == self.hh_size:
```

Comment:
should this be `self.hh_individuals > self.hh_size`?
technically, if there's only 1 individual in the household, then it's within that hh_size so there's no need to increment/change their hh_id yet (and assign them to a new hh?)

Reply:
NVM my comment because i think i understand now that `iter_household` comes after we've assigned the HH ID